app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -6,35 +6,77 @@ module AuthnAzure
     Err = Errors::Authentication::AuthnAzure
     Log = LogMessages::Authentication::AuthnAzure
     # Possible Errors Raised:
-    # TODO: Add errors
+    # TokenFieldNotFoundOrEmpty, MissingRequestParam
 
     Authenticator = CommandClass.new(
       dependencies: {
-        fetch_authenticator_secrets: Authentication::Util::FetchAuthenticatorSecrets.new,
-        logger:                      Rails.logger
+        fetch_authenticator_secrets:   Authentication::Util::FetchAuthenticatorSecrets.new,
+        verify_and_decode_token:       Authentication::AuthnOidc::VerifyAndDecodeToken.new,
```

Comment:
I understand that AuthnOidc and Azure are similar in the way the decoding process is handled. 
IMO involving the logic of another authenticator is confusing because this might not be clear to the next person touching this part of the code. 
I suggest we extract the overlaping logic we need to decode and verify the token for both authenticators and place it in another CommandClass and call this new class both here and in OIDC. That way, the next authenticator that is oauthn2/jwt token based, we will have a more flexible way to perform this action

Reply:
Would you have it in the `Authentication` module? Would you create a new module named `OAuth`?

Reply:
spoken about in person and decided on Oauth